Cannot build for armv7 target

Hello,

I was trying to install [`sysinfo-web`](https://github.com/onur/sysinfo-web) on my Raspberry Pi, but the build of `sysinfo` fails for the armv7 target.

The same errors happen even if I try to build `sysinfo` directly:
```bash
# clone the project
cd /tmp
git clone git@github.com:GuillaumeGomez/sysinfo.git
cd sysinfo

# ensure that the armv7 toolchain is installed
rustup target add armv7-unknown-linux-gnueabihf

# try to build
cargo build --target=armv7-unknown-linux-gnueabihf
```

Here is the output of the build:
```
cargo build --target=armv7-unknown-linux-gnueabihf --verbose
       Fresh libc v0.2.34
   Compiling sysinfo v0.3.17 (file:///tmp/sysinfo)
     Running `rustc --crate-name sysinfo src/sysinfo.rs --crate-type rlib --crate-type cdylib --emit=dep-info,link -C debuginfo=2 -C metadata=2493776c0d9bda08 --out-dir /tmp/sysinfo/target/armv7-unknown-linux-gnueabihf/debug/deps --target armv7-unknown-linux-gnueabihf -C linker=arm-linux-gnueabihf-gcc-4.7 -L dependency=/tmp/sysinfo/target/armv7-unknown-linux-gnueabihf/debug/deps -L dependency=/tmp/sysinfo/target/debug/deps --extern libc=/tmp/sysinfo/target/armv7-unknown-linux-gnueabihf/debug/deps/liblibc-ade5f8b28de429dc.rlib`
error[E0308]: mismatched types
  --> src/linux/disk.rs:63:20
   |
63 |         if statvfs(mount_point_cpath.as_ptr() as *const i8, &mut stat) == 0 {
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
   |
   = note: expected type `*const u8`
              found type `*const i8`

error[E0308]: mismatched types
  --> src/linux/disk.rs:73:22
   |
73 |         total_space: total_space,
   |                      ^^^^^^^^^^^ expected u64, found u32

error[E0308]: mismatched types
  --> src/linux/disk.rs:74:26
   |
74 |         available_space: available_space,
   |                          ^^^^^^^^^^^^^^^ expected u64, found u32

error[E0308]: mismatched types
   --> src/linux/disk.rs:126:24
    |
126 |             if statvfs(mount_point_cpath.as_ptr() as *const i8, &mut stat) == 0 {
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`

error[E0308]: mismatched types
   --> src/linux/disk.rs:127:40
    |
127 |                 self.available_space = stat.f_bsize * stat.f_bavail;
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected u64, found u32

error: aborting due to 5 previous errors

error: Could not compile `sysinfo`.

Caused by:
  process didn't exit successfully: `rustc --crate-name sysinfo src/sysinfo.rs --crate-type rlib --crate-type cdylib --emit=dep-info,link -C debuginfo=2 -C metadata=2493776c0d9bda08 --out-dir /tmp/sysinfo/target/armv7-unknown-linux-gnueabihf/debug/deps --target armv7-unknown-linux-gnueabihf -C linker=arm-linux-gnueabihf-gcc-4.7 -L dependency=/tmp/sysinfo/target/armv7-unknown-linux-gnueabihf/debug/deps -L dependency=/tmp/sysinfo/target/debug/deps --extern libc=/tmp/sysinfo/target/armv7-unknown-linux-gnueabihf/debug/deps/liblibc-ade5f8b28de429dc.rlib` (exit code: 101)
```

Do you know if this can be fixed easily?

Thanks.

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,9 +2,19 @@ language: rust
 matrix:
   include:
     - os: linux
+      env: TARGET=armv7-unknown-linux-gnueabihf
       rust: stable
       dist: trusty
     - os: linux
+      env: TARGET=x86_64-apple-darwin
+      rust: stable
+      dist: trusty
+    - os: linux
+      env: TARGET=armv7-unknown-linux-gnueabihf
+      rust: nightly
+      dist: trusty
+    - os: linux
+      env: TARGET=x86_64-apple-darwin
       rust: nightly
       dist: trusty
     - os: osx

--- a/src/linux/component.rs
+++ b/src/linux/component.rs
@@ -11,7 +11,7 @@ use std::io::Read;
 use std::path::{Path, PathBuf};
 
 /// More information can be found at
-/// http://lxr.free-electrons.com/source/Documentation/hwmon/sysfs-interface
+/// <http://lxr.free-electrons.com/source/Documentation/hwmon/sysfs-interface>
 pub struct Component {
     /// Temperature is in celsius.
     pub temperature: f32,

--- a/src/linux/disk.rs
+++ b/src/linux/disk.rs
@@ -36,8 +36,7 @@ impl From<isize> for DiskType {
     }
 }
 
-fn find_type_for_name(name: &OsStr) -> DiskType
-{
+fn find_type_for_name(name: &OsStr) -> DiskType {
     #![allow(or_fun_call)]
     /* turn "sda1" into "sda": */
     let mut trimmed: &[u8] = name.as_bytes();
@@ -53,6 +52,16 @@ fn find_type_for_name(name: &OsStr) -> DiskType
     DiskType::from(rotational_int.unwrap_or(-1))
 }
 
+#[cfg(target_arch = "arm")]
+macro_rules! cast {
+    ($x:expr) => { $x as u64 }
+}
+
+#[cfg(not(target_arch = "arm"))]
+macro_rules! cast {
+    ($x:expr) => { $x }
+}
+
 pub fn new(name: &OsStr, mount_point: &Path, file_system: &[u8]) -> Disk {
     let mount_point_cpath = utils::to_cpath(mount_point);
     let type_ = find_type_for_name(name);
@@ -60,7 +69,7 @@ pub fn new(name: &OsStr, mount_point: &Path, file_system: &[u8]) -> Disk {
     let mut available_space = 0;
     unsafe {
         let mut stat: statvfs = mem::zeroed();
-        if statvfs(mount_point_cpath.as_ptr() as *const i8, &mut stat) == 0 {
+        if statvfs(mount_point_cpath.as_ptr() as *const _, &mut stat) == 0 {
             total_space = stat.f_bsize * stat.f_blocks;
             available_space = stat.f_bsize * stat.f_bavail;
         }
@@ -70,8 +79,8 @@ pub fn new(name: &OsStr, mount_point: &Path, file_system: &[u8]) -> Disk {
         name: name.to_owned(),
         file_system: file_system.to_owned(),
         mount_point: mount_point.to_owned(),
-        total_space: total_space,
-        available_space: available_space,
+        total_space: cast!(total_space),
+        available_space: cast!(available_space),
     }
 }
 
@@ -123,8 +132,9 @@ impl DiskExt for Disk {
         unsafe {
             let mut stat: statvfs = mem::zeroed();
             let mount_point_cpath = utils::to_cpath(&self.mount_point);
-            if statvfs(mount_point_cpath.as_ptr() as *const i8, &mut stat) == 0 {
-                self.available_space = stat.f_bsize * stat.f_bavail;
+            if statvfs(mount_point_cpath.as_ptr() as *const _, &mut stat) == 0 {
+                let tmp = stat.f_bsize * stat.f_bavail;
+                self.available_space = cast!(tmp);
                 true
             } else {
                 false

